src/testRunner/unittests/config/configurationExtension.ts

```diff
@@ -349,6 +388,40 @@ namespace ts {
                     assert.deepEqual(sourceFile.extendedSourceFiles, expected);
                 });
             });
+
+            describe(testName, () => {
+                it("adds extendedSourceFiles from an array only once", () => {
+                    const sourceFile = readJsonConfigFile("configs/extendsArrayFifth.json", (path) => host.readFile(path));
+                    const dir = combinePaths(basePath, "configs");
+                    const expected = [
+                        combinePaths(dir, "extendsArrayFirst.json"),
+                        combinePaths(dir, "extendsArraySecond.json"),
+                        combinePaths(dir, "extendsArrayThird.json"),
+                        combinePaths(dir, "extendsArrayFourth.json"),
+                    ];
+                    parseJsonSourceFileConfigFileContent(sourceFile, host, dir, {}, "extendsArrayFifth.json");

```

Comment:
You probably want another extendsFile that extends say extendsArrayFirst.json to ensure extendsArrayFirst is listed only once.